Source/inv.cpp

```diff
@@ -888,7 +888,12 @@ void CheckInvCut(int pnum, Point cursorPosition, bool automaticMove, bool dropIt
 	}
 
 	if (dropItem) {
-		TryDropItem();
+		if (TryDropItem()) {
+			// Reset cursor position
+			SetCursorPos(cursorPosition);

```

Comment:
This looks hackish to me. Why does `TryDropItem` change the cursor in the first place?

We should avoid having to "revert" things and instead "just not do them in the first place". Otherwise, the code becomes harder and harder to reason about.

Reply:
TryDropItem doesn't move the cursor, auto equip does. I didn't want to change too much, but I am open to suggestions.

Reply:
@julealgon I probably could check `if (!IsHardwareCursor() && ! dropItem) {` but I prefer having the dropItem feature handled in a single place, instead of spreading the flag.

Reply:
This is bad.... IMHO, we need to rethink this to avoid the cursor change altogether. I'll take a closer look later. Ideally, we'd want to split InvCut into multiple methods, one for each action.

Reply:
I agree with @julealgon
We shouldn't change cursPosition at all. I think that is also not great in the current controller support.

To advance we could (as a first step):

- Add a parameter (for example `Point dropLocation`) to `DropItemBeforeTrig` and `TryInvPut`. And use this instead of `cursPosition`.
- Move `TryDropItem` to `inv.cpp`, cause now it is not only used for controller support.